ipatests/test_integration/test_mkey_upgrade.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright (C) 2015  FreeIPA Contributors see COPYING for license
+#
+
+import re
+import textwrap
+import time
+
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import IntegrationTest
+
+
+class TestMkeyUpgrade(IntegrationTest):
+
+    num_replicas = 1
+    topology = 'line'
+
+    @classmethod
+    def install(cls, mh):
+        cls.master.put_file_contents(
+            '/etc/profile.d/ipaplatform.sh',
+            'export IPAPLATFORM_OVERRIDE=fedora_aes_sha1')
+        cls.master.run_command(['mkdir', '/etc/systemd/system/ipa.service.d'])
+        cls.master.put_file_contents(
+            '/etc/systemd/system/ipa.service.d/platform.conf',
+            '[Service]\n'
+            'Environment="IPAPLATFORM_OVERRIDE=fedora_aes_sha1"')
+        cls.master.run_command(['systemctl', 'daemon-reload'])
+        tasks.install_master(cls.master, setup_dns=False)
+        tasks.install_replica(cls.master, cls.replicas[0], setup_dns=False)
```

Comment:
Shouldn't the platform override also happen on the replica? Or is it that the changes in the platform in this case only matter during initial installation?